### PR TITLE
test(VBtn): add cypress wrapper and better assertions.

### PR DESCRIPTION
**Pull request by JessicaSachs**
## Description
* This change adds a reusable wrapper for `mount` so that `createVuetify` is done under the hood on every `cy.mount` call.
* Additionally, the VBtn tests are refactored to become more meaningful.
* VBtn is also showcased using a presentational test-only grid component (we _could_ use the Vuetify grid system if we wanted to)

### API Changes:
New, more concise syntax for mounting a component with props.

```tsx
import { h } from 'vue'
import VBtn from './VBtn'

it('renders a basic button', () => {
  cy.mount(<VBtn color="success">Let's go!</VBtn>)
})
```

### Visual Changes:
The buttons are properly laid out.

Screen Shot 2021-05-13 at 8 16 02 PM

### API changes:
The previous API for writing a basic test.
```tsx
import { createVuetify } from '@/entry-bundler'
import { mount } from '@cypress/vue'
import { h } from 'vue'
import VBtn from './VBtn'

it('renders a basic button', () => {
  const vuetify = createVuetify()
  mount(VBtn, {
    global: {
      plugins: [vuetify],
    },
    slots: {
      default: 'Button',
    },
  })
})
```

### Visual Changes:
The buttons are smushed together and are not easily viewed or debugged.

Screen Shot 2021-05-13 at 8 20 45 PM

## Motivation and Context:
* By adding a `cy.mount` wrapper, you're able to inject application context like theme providers, i18n, etc.
* Adding styles to the components under test make them easier to debug and view

## How Has This Been Tested?

Using tests.

## Markup:

<details>

```vue
// Paste your FULL Playground.vue here
```
</details>

## Types of changes:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

## Outstanding tasks:
- [x] Fix the type errors
- [x] Refactor the "presentational templates" to be more readable
- [x] Reformat/lint

### CODE REVIEW

**Comment by nekosaur** [comment]
Looks good. I think it would be useful to be able to pass options to `createVuetify` with a mount function like this

```ts
mount: <Props = any>(comp: Component<Props>, options?: CyMountOptions<Props>, vuetifyOptions?: any) => Cypress.Chainable<JQuery<any>>
```

but it looks like CyMountOptions is not exported. although it seems like most things in it are actually flagged as deprecated, so maybe we can just use `MountingOptions` instead?

**Comment by JessicaSachs** [comment]
Alright @nekosaur I think I'm ready for a final review. Any thoughts?

File: packages/vuetify/src/components/VBtn/__tests__/VBtn.spec.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+declare const expect: jest.Expect
```
Comment (nekosaur) [inline packages/vuetify/src/components/VBtn/__tests__/VBtn.spec.ts:1]: Would we have to add this to every non-cypress test file?

File: packages/vuetify/types/cypress.d.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import type { mount as cyMount } from '@cypress/vue'
+
+declare global {
+  namespace Cypress {
+    export interface Chainable {
+      mount: typeof cyMount
```
Comment (nekosaur) [inline packages/vuetify/types/cypress.d.ts:6]: This does not include `vuetifyOptions` argument

File: packages/vuetify/src/components/VBtn/__tests__/VBtn.spec.ts (hunk 1 above)
Comment (JessicaSachs) [reply packages/vuetify/src/components/VBtn/__tests__/VBtn.spec.ts:1]: Yes, if you want the Jest types on the expect matchers. The alternate solution is to pull out the Cypress tests into their own directory. Alternatively you can `import { expect} from 'jest'`.

File: packages/vuetify/types/cypress.d.ts (hunk 2 above)
Comment (JessicaSachs) [reply packages/vuetify/types/cypress.d.ts:6]: I don't really know how to type it. I'll try agian.

File: packages/vuetify/src/components/VBtn/__tests__/VBtn.spec.ts (hunk 1 above)
Comment (nekosaur) [reply packages/vuetify/src/components/VBtn/__tests__/VBtn.spec.ts:1]: I think this works for now.

File: packages/vuetify/types/cypress.d.ts (hunk 2 above)
Comment (nekosaur) [reply packages/vuetify/types/cypress.d.ts:6]: Since we're not actually using the argument yet I think we can revisit it later.

**Review by nekosaur — APPROVED** [review]

**Review by johnleider — APPROVED** [review]

**Pull request merged**